stormpath/resources/custom_data.py

```diff
@@ -71,33 +79,58 @@ def __delitem__(self, key):
 
         for href in self._deletes:
             try:
-                del self.data[href.split('/')[-1]]
+                del self.__dict__.get(self.data_field, {})[href.split('/')[-1]]
             except KeyError:
                 pass
 
-        del self.data[key]
+        del self.__dict__.get(self.data_field, {})[key]
 
         if not self.is_new():
             self._deletes.add(self._get_key_href(key))
 
     def __contains__(self, key):
         self._ensure_data()
-        return key in self.data
+        return key in self.__dict__.get(self.data_field, {})
+
+    def __setattr__(self, name, value):
+        ctype = self.get_resource_attributes().get(name)
+
+        if ctype and not isinstance(value, ctype) \
+                and name not in self.readonly_attrs:
+            getattr(self, name)._set_properties(value)
+        elif name.startswith('_') or name in self.writable_attrs:
+            super(Resource, self).__setattr__(name, value)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Bad first argument 'Resource' given to super()](https://www.codacy.com/app/r/stormpath-sdk-python/file/2393442334/issues/source?bid=3181420&fileBranchId=3230156#l102)